### ISSUE TITLE
[no-extra-parens] Incorrect rule when used with arrow generic

### ISSUE THREAD

**Issue by renjfk** [typescript-eslint/typescript-eslint#1587]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-extra-parens": ["error"]
  }
}
```

```TS
const testFunction = <T>(value: T) => value;
testFunction<() => void>(() => undefined);
```

**Expected Result**
No warning.

**Actual Result**
Warning with
> Unnecessary parentheses around expression.(@typescript-eslint/no-extra-parens)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.0` |
| `@typescript-eslint/parser`        | `2.19.0` |
| `TypeScript`                       | `3.7.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.8.0` |
| `npm`                              | `6.13.7` |

**Comment by nicolashenry** [comment]
I have a similar issue using this:
```ts
const map = new Map<string, () => void>([]);
```
The issue is marked as `has pr` but it seems that the pull request have been closed. Is there a fix ongoing upstream on eslint?

**Comment by bradzacher** [comment]
Not that I know of.
If it's important to you, consider submitting a PR upstream.

**Comment by sviat9440** [comment]
any news?

**Comment by bradzacher** [comment]
With any issue in opened in this project - it either has a visible progress in the form of an attached PR, or it has no progress.

We are a community run project. The **volunteer** maintainers spend most of their time triaging issues and reviewing PRs. This means that most issues *will not progress* unless a member of the community steps up and **champions** it.

If this issue is important to you - consider being that champion.

If not - please just subscribe to the issue and wait patiently.
Commenting asking for status updates does not bump issue priority in any way and just serves to spam ***everyone*** subscribed to the issue.

**Comment by auvred** [comment]
Fixed in https://github.com/typescript-eslint/typescript-eslint/pull/7375

[playground](https://typescript-eslint.io/play/#ts=5.2.2&fileType=.ts&code=MYewdgzgLgBFCm0BiBXMwoEtwwLwwB4AVAPgAoA3AQwBsV4AuGIgSjxJmrvgG4AoBMjQZsYAmTa4OFEJgAm5Cexho58AGaYw8OS36hIsALZUADnhjaA7jACyZgtABOWgOYAaGEqmdZCsgDaALp6QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDWsX7RETVBgDa4bDkTRoHaJBkQAujIC%2BITUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
